test/widgets/message_list_test.dart

```diff
@@ -576,6 +576,39 @@ void main() {
     });
   });
 
+  group('EditStateMarker', () {
+    void checkMarkersCount({required int edited, required int moved}) {
+      check(find.byIcon(ZulipIcons.edited).evaluate()).length.equals(edited);
+      check(find.byIcon(ZulipIcons.message_moved).evaluate()).length.equals(moved);
+    }
+
+    testWidgets('no edited or moved messages', (tester) async {
+      final message = eg.streamMessage();
+      await setupMessageListPage(tester, messages: [message]);
+      checkMarkersCount(edited: 0, moved: 0);
+    });
+
+    testWidgets('edited and moved messages from events', (tester) async {
```

Comment:
Let's have another test case that checks that the markers don't affect the layout of anything else. That seems like a class of bug that could easily be introduced in some future change to the message UI.

So that can look like:
* put three messages on the screen; the one we'll change the state of, and one each before and after it
* note the Rect position of everything relevant:
  * each of the three messages
  * the target message's star (make it starred so this appears)
  * some text in the target message
  * maybe other things I'm not thinking of
  * … oh a reaction, that's a good one (and there might still be more I'm not thinking of)
* then make the message be moved; and look at all the same locations and confirm none have changed
* then make the message be edited; and ditto

Reply:
Added the "edit state updates do not affect layout" test case. It uses a heavy helper function to record the items you mentioned, as well as `ImageNode`s and `ParagraphNode`s from the target message.